watsononlinestore/watson_online_store.py

```diff
@@ -72,17 +72,23 @@ def context_merge(self, dict1, dict2):
 
         return new_dict
 
-    def parse_slack_output(self, slack_rtm_output):
-        output_list = slack_rtm_output
+    def parse_slack_output(self, output_list):
         if output_list and len(output_list) > 0:
             for output in output_list:
-                if output and 'text' in output and \
-                    'user_profile' not in output and \
-                        self.at_bot in output['text']:
-                    return (output['text'].split(
-                        self.at_bot)[1].strip().lower(),
-                         output['channel'],
-                         output['user'])
+                if output and 'text' in output and (
```

Comment:
the commit message had such a nice explanation about the logic being done here, why not add that in the form of comments :)